homeassistant/components/roku/select.py

```diff
@@ -0,0 +1,173 @@
+"""Support for Roku selects."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+
+from rokuecp import Roku
+from rokuecp.models import Device as RokuDevice
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import roku_exception_handler
+from .const import DOMAIN
+from .coordinator import RokuDataUpdateCoordinator
+from .entity import RokuEntity
+from .helpers import format_channel_name
+
+
+@dataclass
+class RokuSelectEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    options_fn: Callable[[RokuDevice], list[str]]
+    value_fn: Callable[[RokuDevice], str | None]
+    set_fn: Callable[[RokuDevice, Roku, str], Awaitable[None]]
+
+
+def _get_application_name(device: RokuDevice) -> str | None:
+    if device.app is None or device.app.name is None:
+        return None
+
+    if device.app.name == "Roku":
+        return "Home"
+
+    return device.app.name
+
+
+def _get_applications(device: RokuDevice) -> list[str]:
+    return ["Home"] + sorted(app.name for app in device.apps if app.name is not None)
+
+
+def _get_channel_name(device: RokuDevice) -> str | None:
+    if device.channel is None:
+        return None
+
+    return format_channel_name(device.channel.number, device.channel.name)
+
+
+def _get_channels(device: RokuDevice) -> list[str]:
+    return sorted(
+        format_channel_name(channel.number, channel.name) for channel in device.channels
+    )
+
+
+async def _launch_application(device: RokuDevice, roku: Roku, value: str) -> None:
+    if value == "Home":
+        await roku.remote("home")
+
+    appl = next(
+        (app for app in device.apps if value == app.name),
+        None,
+    )
+
+    if appl is not None and appl.app_id is not None:
+        await roku.launch(appl.app_id)
+
+
+async def _tune_channel(device: RokuDevice, roku: Roku, value: str) -> None:
+    _channel = next(
+        (
+            channel
+            for channel in device.channels
+            if (
+                channel.name is not None
+                and value == f"{channel.name} ({channel.number})"
+            )
+            or value == channel.number
```

Comment:
Can you use `format_channel_name` here?

Reply:
I had tried once it was throwing some syntax issue, could of been unrelated I'll try again